src/tests/nativeaot/GenerateUnmanagedEntryPoints/GenerateUnmanagedEntryPoints.csproj

```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <CLRTestKind>BuildAndRun</CLRTestKind>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```

Comment:
I think you need to add `<IlcExportUnmanagedEntrypoints>true</IlcExportUnmanagedEntrypoints>` here to make the test work.

Reply:
Unrelated: It would be nice to change the line that sets the default in `Microsoft.NETCore.Native.targets` to 
```
<IlcExportUnmanagedEntrypoints Condition="$(IlcExportUnmanagedEntrypoints) == '' and '$(NativeLib)' == 'Shared'">true</IlcExportUnmanagedEntrypoints>
```
to allow overriding it.

Reply:
Thank you, I missed the part that the `Exports` file is used for native linking and that it is not generated when the application is compiled as an executable.

Reply:
The test is still failing on `linux-x64`
I have checked the build if the `Exports` file is passed to the linker.
It seems that is correct, however the test still fails to load the symbols:
```
"clang"
"/__w/1/s/artifacts/tests/coreclr/obj/linux.x64.Debug/Managed/nativeaot/GenerateUnmanagedEntryPoints/GenerateUnmanagedEntryPoints/native/GenerateUnmanagedEntryPoints.o"
-o
"/__w/1/s/artifacts/tests/coreclr/linux.x64.Debug/nativeaot/GenerateUnmanagedEntryPoints/GenerateUnmanagedEntryPoints/native/GenerateUnmanagedEntryPoints"
-Wl,--version-script=/__w/1/s/artifacts/tests/coreclr/obj/linux.x64.Debug/Managed/nativeaot/GenerateUnmanagedEntryPoints/GenerateUnmanagedEntryPoints/native/GenerateUnmanagedEntryPoints.exports
/__w/1/s/artifacts/bin/coreclr/linux.x64.Debug/aotsdk/libbootstrapper.a
/__w/1/s/artifacts/bin/coreclr/linux.x64.Debug/aotsdk/libRuntime.WorkstationGC.a
/__w/1/s/artifacts/bin/coreclr/linux.x64.Debug/aotsdk/libeventpipe-disabled.a
/__w/1/s/artifacts/bin/coreclr/linux.x64.Debug/aotsdk/libstdc++compat.a
/__w/1/s/artifacts/bin/coreclr/linux.x64.Debug/aotsdk/libnumasupportdynamic.a
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libSystem.Native.a
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libSystem.Globalization.Native.a
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libSystem.IO.Compression.Native.a
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libSystem.Net.Security.Native.a
/__w/1/s/artifacts/bin/microsoft.netcore.app.runtime.linux-x64/Release/runtimes/linux-x64/native/libSystem.Security.Cryptography.Native.OpenSsl.a
-g
-Wl,-rpath,'$ORIGIN/..'
-Wl,--build-id=sha1
-Wl,--as-needed
-pthread
-ldl
-lz
-lrt
-lm
-pie
-Wl,-pie
-Wl,-z,relro
-Wl,-z,now
-Wl,--discard-all
-Wl,--gc-sections
``` 
I don't have access to a `linux-x64` machine at the moment, but I will continue investigating why this is failing.

Reply:
Fixed with: https://github.com/dotnet/runtime/pull/83396#discussion_r1141882182